File: baby_back/src/main/java/com/backend/quest/domain/Quest.java
package com.backend.quest.domain;

import lombok.*;

@Getter
@ToString
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Quest {

    private Long questId;
    private String title;
    private String description;
    private String type;       // DAILY | URGENT
    private String difficulty; // EASY | MEDIUM | HARD
    private String theme;      // CARE | ACTIVITY | EMOTION | REQUEST
    private int reward;
    private int urgency;
    private int dueDays;
    private boolean active;
}

File: baby_front/src/types/quest.ts
export type QuestType = "DAILY" | "WEEKLY" | "EVENT" | "URGENT";
export type QuestStatus = "TODO" | "DONE" | "FAILED" | "EXPIRED";

export interface Quest {
  questId: number;
  title: string;
  description?: string;
  type: QuestType;
  reward: number;
  urgency?: number;
  difficulty?: string;
  theme?: string;
  dueDays?: number;
}

export interface MemberQuest {
  id: number;
  memberEmail?: string;
  questId: number;
  status: QuestStatus;
  assignedDate: string;
  completedAt: string | null;
  dueDate?: string | null;
  quest: Quest;
}

export interface QuestHome {
  dailyQuests: MemberQuest[];
  weeklyQuests: MemberQuest[];
  eventQuests: MemberQuest[];
  urgentQuests: MemberQuest[];
  point: number;
}
